### ISSUE TITLE
Show operation addClause ignoring for SoftDeletes

### ISSUE THREAD

**Issue by skater4** [Laravel-Backpack/CRUD#5204]
# Bug report

### What I did

any crud controller, setup method
`$this->crud->addClause('whereIn', 'feed_id', [1,2,3]);` (any condition)

### What I expected to happen

show operation should return 404 if current entity doesnt have feed_id in 1,2,3

### What happened

if model has SoftDeletes - it return entity page

### Is it a bug in the latest version of Backpack?

5.6.1

### After I run ```composer update backpack/crud``` the bug... is it still there?

Yes

### Backpack, Laravel, PHP, DB version

When I run ```php artisan backpack:version``` the output is:

### PHP VERSION:
PHP 8.2.6 (cli) (built: May 23 2023 09:42:25) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.2.6, Copyright (c) Zend Technologies

### LARAVEL VERSION:
v9.39.0@67e674709e1e7db14f304a871481f310822d68c5

### BACKPACK PACKAGE VERSIONS:
backpack/crud: 5.6.1
backpack/generators: 3.3.16
backpack/permissionmanager: 6.0.17

**Comment by tabacitu** [comment]
Hi @skater4 ,

Indeed, the default behaviour is that... if SoftDeletes is present on the Model, the Show operation WILL show that entry and add a `soft_deletes` column. 

You can change that behaviour in the config file for that operation. Just go to `config/backpack/operations/show.php` and change this to false:

```php
    // If model has SoftDeletes, allow the admin to access the Show page for
    // soft deleted items & add a deleted_at column to ShowOperation?
    'softDeletes' => false,
```

Let me know if that doesn't work for you.

Cheers!

**Comment by skater4** [comment]
> Hi @skater4 ,
> 
> Indeed, the default behaviour is that... if SoftDeletes is present on the Model, the Show operation WILL show that entry and add a `soft_deletes` column.
> 
> You can change that behaviour in the config file for that operation. Just go to `config/backpack/operations/show.php` and change this to false:
> 
> ```
>     // If model has SoftDeletes, allow the admin to access the Show page for
>     // soft deleted items & add a deleted_at column to ShowOperation?
>     'softDeletes' => false,
> ```
> 
> Let me know if that doesn't work for you.
> 
> Cheers!

i think u didnt understand me. soft deletes are ok, BUTTT
for example there are rows like this
id feed_id
1  10
2  20
3  30
and i make in setup()
$this->crud->addClause('whereIn', 'feed_id', [10,20]);
so list operation will show only ids 1 and 2
so by logic show operation should also show only ids 1 and 2 and id =  3 return 404
but it returns entity page that not allow me do any access restrictions for models with SoftDeletes

**Comment by tabacitu** [comment]
Oh ok I think I understand now. Let me rephrase:

In your `setup()` method you did `$this->crud->addClause('whereIn', 'feed_id', [10,20]);`, and expected this restriction to apply to ALL operations. But that didn't happen. It applied to the ListOperation, but not to the ShowOperation.

---

That's true, ShowOperation doesn't use the same general `$this->crud->query` so your `addClause()` statements won't apply there. It just does a `findOrFail()`. But... that's the case with ALL entry-level operations. Your statement will not work on UpdateOperation or DeleteOperation either. To customize the query on entry-level operations, you can target those operations individually and prevent them from working:

```php
        // instead of this
        $this->crud->addClause('whereIn', 'feed_id', [10,20]);

        // you should do something like this
        $this->crud->operation(['show', 'update'], function() {
            $id = $this->crud->getCurrentEntryId();
            // TODO: now check that the ID is allowed to be updated/edited, otherwise abort
        });
```

I do not recommend editing something general like the query in `setup()` directly. Because it applies to ALL operations, and not all operations will behave the same. It is much better to do your operation customizations either in the `setupXxxOperation()` methods, or in `setup()` but in closures that make it clear WHAT operation should have that customization, like instructed above.

--- 

The question now becomes... and I'm asking you @pxpm ... SHOULD WE change how we get the current entry... in order to use `$this->crud->query`? What do you think?

**Comment by skater4** [comment]
> Oh ok I think I understand now. Let me rephrase:
> 
> In your `setup()` method you did `$this->crud->addClause('whereIn', 'feed_id', [10,20]);`, and expected this restriction to apply to ALL operations. But that didn't happen. It applied to the ListOperation, but not to the ShowOperation.
> 
> That's true, ShowOperation doesn't use the same general `$this->crud->query` so your `addClause()` statements won't apply there. It just does a `findOrFail()`. But... that's the case with ALL entry-level operations. Your statement will not work on UpdateOperation or DeleteOperation either. To customize the query on entry-level operations, you can target those operations individually and prevent them from working:
> 
> ```
>         // instead of this
>         $this->crud->addClause('whereIn', 'feed_id', [10,20]);
> 
>         // you should do something like this
>         $this->crud->operation(['show', 'update'], function() {
>             $id = $this->crud->getCurrentEntryId();
>             // TODO: now check that the ID is allowed to be updated/edited, otherwise abort
>         });
> ```
> 
> I do not recommend editing something general like the query in `setup()` directly. Because it applies to ALL operations, and not all operations will behave the same. It is much better to do your operation customizations either in the `setupXxxOperation()` methods, or in `setup()` but in closures that make it clear WHAT operation should have that customization, like instructed above.
> 
> The question now becomes... and I'm asking you @pxpm ... SHOULD WE change how we get the current entry... in order to use `$this->crud->query`? What do you think?

Thank you for reply.
"But that didn't happen" - if model has soft deletes - yes, it works only for list, but not show. 
But if model DOESN'T have soft deletes - addClause works for ALL operations that breaks logic. I think soft deletes should not affect this logic

**Comment by pxpm** [comment]
> The question now becomes... and I'm asking you @pxpm ... SHOULD WE change how we get the current entry... in order to use $this->crud->query? What do you think?

@tabacitu I've worked on this in: https://github.com/Laravel-Backpack/CRUD/pull/4937 https://github.com/Laravel-Backpack/CRUD/pull/4555

**Comment by tabacitu** [comment]
So what do you recommend we do @pxpm ?

**Comment by skater4** [comment]
Maybe u make some fix with backward compatibility? Possible defining deprecated old functionality

**Comment by pxpm** [comment]
I've submitted #5271 to address this issue in a non-breaking way. Let's continue any conversation over there. 

@skater4 let me know if you think the solution would work for you or you may have a better one 🙏

**Comment by skater4** [comment]
> I've submitted #5271 to address this issue in a non-breaking way. Let's continue any conversation over there.
> 
> @skater4 let me know if you think the solution would work for you or you may have a better one pray

Yes, usePanelQuery parameter would be good for now. But any case SoftDeletes behavior still confusing and non logical
